### DISCUSSION TITLE
net.core 5  &Orchard core Whether to consider supporting dynamic compilation of template resources, views, styles, JS and other files?

### DISCUSSION THREAD

**Discussion by pgy866** [OrchardCMS/OrchardCore#10419] [Ideas]
https://github.com/OrchardCMS/OrchardCore/tree/5476775f5f515872b3632c2388f51ff68e15a73f/src/OrchardCore.Themes/TheBlogTheme/wwwroot

Template resource files, layout, style and other resources required by the template. Now it has NET5.0, Whether to consider supporting dynamic compilation of template resources, views, styles, JS and other files?

模板的资源文件及布局、样式等模板所需资源。现在已经NET5.0，是否考虑支持动态编译？而不是封装到DLL。

**Comment by Skrypt** [comment]
You mean like gulp watch? This will work only if you modify the bootstrap-oc.scss files with `gulp watch`. For the other assets, the gulp pipeline would require to compile the content in the src folder of the assets to add a watch on those files. Never tried yet.

**Comment by pgy866** [comment]
Not this!
I just don't want to compile js, css, image …resource files into dll.
非也！
我只是不想把js     css    image……  文件编译到到dll里面。

**Reply by Skrypt** [reply]
You could do this for your own themes by using the StaticFileProvider and override assets in your Web project inside the wwwroot folder. Though, since that web project can be different for everyone then it's on you to build it that way by adding an Assets.json file to your Web project I guess. So basically /wwwroot/tenantname/somepathto/somefile.css

**Reply by pgy866** [reply]
Thank you, it's a good method, but for out-of-the-box users (non-programmers) of CMS, it doesn't seem to be very friendly

谢谢，是一个不错的方法，但是针对CMS的开箱即用的用户（非程序员）来说，它好像并不是很友好

**Reply by Skrypt** [reply]
Yes, then in this case we would need to be able to rebuild assets from server side when a change is made in a scss or .js file from the admin UI. But then, it would involve requiring some programmation knowledge. So, I think this has been requested in other issues. I understand your request. Though, there is nothing currently that I know planned about this. Keeping this issue open for further discussion development.

**Comment by RockNHawk** [comment]
If endpoint user buy a CMS from a company ( no source code ), the CMS is developed by Orchard Core.
endpoint user may have some basic `css` `js` skill, he want's to modifly the style, currently the Orchard Core's default behavior does not supported.

**Reply by pgy866** [reply]
This is not a good thing for the reuse and sharing of the "templates" of "ORCHARD CORE". It may prevent the sharing of ORCHARD CORE's templates, such as the huge number of out-of-the-box WEB front-end templates like "WordPress". Sharing from the community。
`这对于“ORCHARD CORE”的“模板”复用及分享不是一件好事，他可能会阻止ORCHARD CORE的模板分享，比如：“WordPress”那样庞大数量的开箱即可用的WEB前端模板来自社区的分享`

**Reply by Skrypt** [reply]
We have recipes that can be shared that can contain `scriptblock` liquid tags.
As for sharing, I think it's not an issue. It's more about being able to transpile SCSS files on runtime or transpile TypeScript on runtime or not.

**Comment by Skrypt** [comment]
It is possible now with Dart-Sass to compile .scss files from runtime if we would compile it with the javascript version of Dart-Sass. Though for javascript, it is probably not recommended to recompile for example an Angular app on runtime.

So, for javascript you can actually add files in the Media Gallery and reference them from your Liquid templates. Same can be done with .css files. Though, what will probably be asked, is to be able to transpile SASS from runtime.
